File: src/types/project.ts
export type ProjectStatus = "active" | "archived";

export type Project = {
  id: string;
  name: string;
  description?: string | null;
  repoPath?: string | null;
  githubUrl?: string | null;
  projectType?: string | null;
  workspaceId?: string | null;
  workspaceRelativePath?: string | null;
  status: ProjectStatus;
  createdAt: string;
  updatedAt: string;
};

export type CreateProjectInput = {
  name: string;
  description?: string | null;
  repoPath?: string | null;
  githubUrl?: string | null;
  projectType?: string | null;
};

export type UpdateProjectInput = Partial<CreateProjectInput> & {
  status?: ProjectStatus;
};

export type ProjectStats = {
  projectId: string;
  projectName: string;
  commitsThisWeek: number;
  lastSync?: string | null;
  hoursTracked: number;
};

export type CategoryDistribution = {
  category: string;
  count: number;
  percentage: number;
};

export type RecentCommit = {
  projectId: string;
  projectName: string;
  repoPath?: string | null;
  commitHash: string;
  message: string;
  authorName?: string | null;
  branch?: string | null;
  committedAt: string;
  refs: CommitRefSummary[];
  worktree?: CommitWorktreeSummary | null;
  status: string;
};

export type TopContributor = {
  authorName: string;
  authorEmail?: string | null;
  commitCount: number;
};

export type GitBranch = {
  name: string;
  kind: "local" | "remote";
  isCurrent: boolean;
};

export type GitRef = {
  projectId: string;
  name: string;
  fullName: string;
  kind: "local" | "remote";
  isCurrent: boolean;
  isHead: boolean;
  lastSeenCommit?: string | null;
  lastScannedAt: string;
};

export type GitRefFilter = {
  projectId?: string | null;
  name: string;
  kind: "local" | "remote";
};

export type GitWorktree = {
  projectId: string;
  path: string;
  branch?: string | null;
  headCommit?: string | null;
  isClean?: boolean | null;
  isPrunable: boolean;
  isLocked: boolean;
  lastScannedAt: string;
};

export type CommitRefSummary = {
  name: string;
  kind: "local" | "remote";
  isCurrent: boolean;
};

export type CommitWorktreeSummary = {
  path: string;
  branch?: string | null;
  headCommit?: string | null;
  isClean?: boolean | null;
};

export type ProjectGitFocus = {
  projectId: string;
  refs: GitRefFilter[];
  worktreePaths: string[];
};

export type SaveProjectGitFocusInput = ProjectGitFocus;

File: src-tauri/src/domain/project.rs
use serde::{Deserialize, Serialize};

use crate::domain::git_metadata::{CommitRefSummary, CommitWorktreeSummary};

#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Project {
    pub id: String,
    pub name: String,
    pub description: Option<String>,
    pub repo_path: Option<String>,
    pub github_url: Option<String>,
    pub github_account_id: Option<String>,
    pub project_type: Option<String>,
    pub workspace_id: Option<String>,
    pub workspace_relative_path: Option<String>,
    pub classification: String,
    pub status: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateProjectInput {
    pub name: String,
    pub description: Option<String>,
    pub repo_path: Option<String>,
    pub github_url: Option<String>,
    pub github_account_id: Option<String>,
    pub project_type: Option<String>,
    pub classification: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateProjectInput {
    pub name: Option<String>,
    pub description: Option<String>,
    pub repo_path: Option<String>,
    pub github_url: Option<String>,
    pub github_account_id: Option<String>,
    pub project_type: Option<String>,
    pub workspace_id: Option<String>,
    pub workspace_relative_path: Option<String>,
    pub classification: Option<String>,
    pub status: Option<String>,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ProjectStats {
    pub project_id: String,
    pub project_name: String,
    pub commits_this_week: i64,
    pub last_sync: Option<String>,
    pub hours_tracked: f64,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CategoryDistribution {
    pub category: String,
    pub count: i64,
    pub percentage: f64,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RecentCommit {
    pub project_id: String,
    pub project_name: String,
    pub repo_path: Option<String>,
    pub commit_hash: String,
    pub message: String,
    pub author_name: Option<String>,
    pub branch: Option<String>,
    pub committed_at: String,
    pub refs: Vec<CommitRefSummary>,
    pub worktree: Option<CommitWorktreeSummary>,
    pub status: String,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct TopContributor {
    pub author_name: String,
    pub author_email: Option<String>,
    pub commit_count: i64,
}
